src/amber/models/granite.cr

```diff
@@ -0,0 +1,16 @@
+module Amber::Models::Granite
+  macro included
+	  def initialize
+	  end
+
+	  def initialize(args : Hash(String | Symbol, Amber::Router::ParamsType))
+        set_attributes(args)
+  	  end
+
+	  def set_attributes(args : Hash(Symbol | String, Amber::Router::ParamsType))
+        args.each do |k, v|
+          cast_to_field(k, v.as(DB::Any))
```

Comment:
instead of doing this here, can we cast it in the controller before passing into Granite?  Also, how does Crecto handle this?

Reply:
Well, the controller params does not care about the Model. The model should be responsible for casting to what it needs. 

Also the `cast_to_field` also live in Granite::ORM. We can easily just move the initialize and set_attributes to Granite

Reply:
I would prefer not requiring users to remember they need the included this module.  Can we move this to Granite?